examples/dbscan/dbscan.cpp

```diff
@@ -154,12 +156,30 @@ int main(int argc, char *argv[])
         "Testing::compute_centers",
         Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_clusters),
         KOKKOS_LAMBDA(int const i) {
+          // The only reason we use heap here is for reproducibility. Current
+          // DBSCAN algorithm does not guarantee that the indices corresponding
+          // to the same cluster are going to appear in the same order from run
+          // to run. Using heap, we explicitly sort them in ascending order,
+          // thus guaranteeing the same summation order when computing cluster
+          // centers.
+          auto *cluster_start = cluster_indices.data() + cluster_offset(i);
           int cluster_size = cluster_offset(i + 1) - cluster_offset(i);
 
+          // Initialize heap
+          for (int j = 0; j < cluster_size; ++j)
+            ArborX::Details::pushHeap(cluster_start, cluster_start + j + 1,
+                                      ArborX::Details::Greater<int>());
+
           ArborX::Point cluster_center{0.f, 0.f, 0.f};
-          for (int j = cluster_offset(i); j < cluster_offset(i + 1); j++)
+          for (int j = 0; j < cluster_size; ++j)
           {
-            auto const &cluster_point = primitives(cluster_indices(j));
+            // Pop heap
+            ArborX::Details::popHeap(cluster_start,
+                                     cluster_start + cluster_size - j,
+                                     ArborX::Details::Greater<int>());
+            auto index = *(cluster_start + cluster_size - 1 - j);
+
+            auto const &cluster_point = primitives(index);
```

Comment:
Did you measure that this performs better than doing a regular heap sort and then loop (in the same kernel) and compute centers like you used to?

Reply:
No, I did not check the performance. I only checked that it runs reasonably fast.

Reply:
If there is no good reason not to do so, sort first.  This way you could abstract the heap sort away which would improve readability.

Reply:
I don't understand what you are saying. What is regular heap sort? What loop? Can you write the piece of code?

Reply:
Addressed by introducing `heapSort`, and using it here.